make type error for ARM aarch64 u8/i8

From https://github.com/pola-rs/r-polars/pull/377#issuecomment-1731195728

```log
 error[E0308]: mismatched types
   --> /home/runner/.cargo/git/checkouts/extendr-f64cf2fdda5197e8/ee8d375/extendr-api/src/lib.rs:630:17
    |
630 |         Rprintf(PRINTF_NO_FMT_CSTRING.as_ptr(), cs.as_ptr());
    |         ------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*const u8`, found `*const i8`
    |         |
    |         arguments to this function are incorrect
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`
note: function defined here
   --> /home/runner/work/r-polars/r-polars/src/rust/target/aarch64-unknown-linux-gnu/release-optimized/build/libR-sys-d7c4c89c6e4571b8/out/bindings.rs:876:12
    |
876 |     pub fn Rprintf(arg1: *const ::std::os::raw::c_char, ...);
    |            ^^^^^^^

error[E0308]: mismatched types
   --> /home/runner/.cargo/git/checkouts/extendr-f64cf2fdda5197e8/ee8d375/extendr-api/src/lib.rs:638:18
    |
638 |         REprintf(PRINTF_NO_FMT_CSTRING.as_ptr(), cs.as_ptr());
    |         -------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `*const u8`, found `*const i8`
    |         |
    |         arguments to this function are incorrect
    |
    = note: expected raw pointer `*const u8`
               found raw pointer `*const i8`
note: function defined here
   --> /home/runner/work/r-polars/r-polars/src/rust/target/aarch64-unknown-linux-gnu/release-optimized/build/libR-sys-d7c4c89c6e4571b8/out/bindings.rs:879:12
    |
879 |     pub fn REprintf(arg1: *const ::std::os::raw::c_char, ...);
    |            ^^^^^^^^
```

After #566, extendr doen't seem able to built for aarch64.
Please see remacs/remacs#1393 for details.

--- a/extendr-api/src/lib.rs
+++ b/extendr-api/src/lib.rs
@@ -621,7 +621,7 @@ pub fn sxp_to_rtype(sxptype: i32) -> Rtype {
     }
 }
 
-const PRINTF_NO_FMT_CSTRING: &[i8] = &[37, 115, 0]; // same as "%s\0"
+const PRINTF_NO_FMT_CSTRING: &[std::os::raw::c_char] = &[37, 115, 0]; // same as "%s\0"
 #[doc(hidden)]
 pub fn print_r_output<T: Into<Vec<u8>>>(s: T) {
     let cs = CString::new(s).expect("NulError");

